packages/server/src/user/__tests__/user.controller.spec.ts

```diff
@@ -0,0 +1,166 @@
+    dummyUsers = await userTestUtil.createDummyUsers();
+
+    // create controller and service objects
+    userService = new UserService(prisma);
+    userController = new UserController(userService);
+  });
+
+  afterAll(async () => {
+    await userTestUtil.tearDown();
+  });
+
+  beforeEach(async () => {
+    randomProject = dummyProjects[Math.floor(Math.random() * dummyProjects.length)];
+    randomUser = dummyUsers.concat(dummyAdmins)[Math.floor(Math.random() * (dummyAdmins.length + dummyUsers.length))];
+  });
+
+  describe('Test cases for getMyInfo() API', () => {
+    it('Return requested user object (self)', async () => {
+      const req = { user: { id: randomUser.id } };
+      const responseUser = await userController.getMyInfo(req);
+
+      expect(responseUser).toEqual(randomUser);
+    });
+
+    it('Throw an error for requesting a non-existing user (self)', async () => {
+      const req = { user: { id: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' } };
+      expect(userController.getMyInfo(req)).rejects.toThrow(HttpException);
+    });
+  });
+
+  describe('Test cases for getAllUsersFromCurrentProject() API', () => {
+    it('Return requested users for a project', async () => {
+      const req = { user: { projectId: randomProject.id } };
+      const responseUsers = await userController.getAllUsersFromCurrentProject(req);
+      const matchUsers = dummyUsers.concat(dummyAdmins).filter((element) => {
+        return element.projectId === randomProject.id;
+      });
+
+      for (const user of matchUsers) {
+        expect(responseUsers).toContainEqual(user);
+      }
+    });
+
+    it('Return empty list for requesting a non-existing project', async () => {
+      const req = { user: { projectId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' } };
+      expect(userController.getAllUsersFromCurrentProject(req)).resolves.toEqual([]);
+    });
+  });
+
+  describe('Test cases for getUserInfo() API', () => {
+    it('Return requested user object (other)', async () => {
+      const responseUser = await userController.getUserInfo(randomUser.id);
+      expect(responseUser).toEqual(randomUser);
+    });
+
+    it('Throw an erorr for requesting a non-existing user (other)', async () => {
+      const userId = 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx';
+      expect(userController.getUserInfo(userId)).rejects.toThrow(HttpException);
+    });
+  });
+
+  describe('Test cases for addRoleToUser() API', () => {
+    it('Add new roles to user should success', async () => {
+      const randomNormalUser = dummyUsers[Math.floor(Math.random() * dummyUsers.length)];
+      const rolesToAdd = [parseInt('0001', 2), parseInt('0100', 2)]; // Add admin role '1' and custom role '4'
+
+      const responseVal = await userController.addRoleToUser(randomNormalUser.id, rolesToAdd[0] + rolesToAdd[1]);
+      const userEdited = await prisma.user.findUnique({ where: { id: randomNormalUser.id } });
+
+      expect(responseVal).toBe(true);
+
```

Comment:
This should also expect, same for all role check with valid users
```suggestion
expect(userEdited.role).toBe(5);
```

Reply:
Added